Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -2175,6 +2176,105 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncToSpliced, (JSGlobalObject* globalObject,
     return JSValue::encode(result);
 }
 
+static uint64_t flatIntoArray(JSGlobalObject* globalObject, JSObject* target, JSObject* source, uint64_t sourceLength, uint64_t targetIndex, uint64_t depth)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
```

Comment:
We need to have recursion check via `vm.isSafeToRecurseSoft`. `flatIntoArray` is recursively calling, so it can be possible that we will have stackoverflow.